internal/buffer/settings.go

```diff
@@ -50,8 +52,16 @@ func (b *Buffer) DoSetOptionNative(option string, nativeValue interface{}) {
 		return
 	}
 
+	oldValue := b.Settings[option]
 	b.Settings[option] = nativeValue
 
+	if oldValue != nativeValue {
```

Comment:
Any reason why do this before all the below stuff? Shouldn't we rather call `onBufferOptionChanged` at the end, after the option has been fully set?